### DISCUSSION TITLE
Question about getServerSideProps loading with local data

### DISCUSSION THREAD

**Discussion by ben-me** [neuefische/questions#225] [Web] [answered]
Heyo,

I'm still working with locally saved data in my nextJS app. Just created a page to see my reservations.
When routing to the page, the data that is in the local "database" (hardcoded) is rendered properly.

If I go and add/save a new reservation (not yet persistent since I work with local data) and go back to the reservation page, the new entry isn't selected or rendered. 

[NextJS docs](https://nextjs.org/docs/basic-features/data-fetching/get-server-side-props) tells us that

> getServerSideProps

> If you export a function called getServerSideProps (Server-Side Rendering) from a page, Next.js will pre-render this page on each request using the data returned by getServerSideProps.

As far as I understand the request is the part when I route back to the reservation page. Shouldn't the data be read again when going there and in turn display the newly created entry?

This is my getServerSideProps function
```jsx
export async function getServerSideProps() {
  const today = new Date();
  const givenDate = today.toISOString().substring(0, 10);
  const foundReservations = await getReservationByUser(givenDate);
  return {
    props: {
      foundReservations,
    },
  };
}
```

And this is how I read it:

```jsx
export function getReservationByUser(date) {
  const userReservations = [];
  rooms.map((room) => {
    room.desks.map((desk) => {
      desk.reservations.map((reservation) => {
        const today = Date.parse(date);
        const resDate = Date.parse(reservation.date);
        if (reservation.user === 'user1' && resDate > today) {
          reservation.room = room.name;
          reservation.desk = desk.name;
          userReservations.push(reservation);
        }
      });
    });
  });
  userReservations.sort((a, b) => {
    a = a.date.split('-').join('');
    b = b.date.split('-').join('');
    return a.localeCompare(b);
  });

  return userReservations;
}
```

I'm not sure if this problem will be solved after switching to a DB but right now I want to have the locally added entry be shown on the reservation page.

**Comment by pixelass** [comment]
The data is never stored. It is temporarily stored and then discarded. Remember how I mentioned that in the other question concerning the SSR/SSG integration of useSWR. 

If you want to store it you have to use a database or `localStorage` (while `localStorage` is not relevant for SSR/SSG since it happens client side).

As mentioned before you are using it in a way, it's not intended to be used, therefore it cannot work (or has flaws).

**Comment by ben-me** [comment]
I know that its only temporarily stored (although I'm not quite sure where since I don't use localStorage - maybe cache?) and that's my big problem right now.

Maybe I should have switched to a DB earlier but I wanted to see if everything works like this first.

useSWR worked on the other page even though it is not intended to be used like that.
On this Page though I can't manage to misuse it for my purpose. 


My current user story revolves around displaying the reservations. So what should I do now? It's possible to render the stuff that in the local file but nothing else.

**Comment by ben-me** [comment, marked as answer]
I discussed this specific situation with my coaches and we agreed to do the following:

- Instead of using SWR or anything else I will add an additional overarching useState which will be filled with my locally saved data
- This useState will be the basis for all CRUD Operations (adding, deleting, rendering etc.)

This will hopefully be enough to pass all the acceptance criteria of my current user story.
Afterwards I will switch to a Database and exchange the useState for useSWR to have some client side fetching.
